index.js

```diff
@@ -26,6 +29,50 @@ function escapeMessage(text) {
   );
 }
 
+/**
+ *
+ * @param {String} channelId | ID of channel to check/join
+ * @param {String} token | slack token for which to check channel membership
+ */
+async function checkJoinChannel({ channelId, token }) {
+  try {
+    const res = await app.client.conversations.info({
+      token: token,
+      channel: channelId,
+      include_num_members: false,
+      include_locale: false
+    });
+
+    //check for a bad response & throw it
+    if (!res.ok) {
+      //check if the channel wasn't found
+      if (res.error === 'channel_not_found') {
+        // it couldn't find the channel, so we'll return false
+        // a false return indicates later that it'll send via webhook
+        return false;
+      }
+      // if it isn't returning false, it'll throw an error
+      throw res;
+    }
+
+    if (!res.channel.is_member) {
+      await app.client.conversations.join({
+        channel: channelId,
+        token: token
+      });
+      return true;
+    }
+
+    if (res.channel.is_member) {
+      return true;
+    }
```

Comment:
Let's make this an if/else.

```suggestion
    if (res.channel.is_member) {
      return true;
    } else {
      await app.client.conversations.join({
        channel: channelId,
        token: token
      });
      return true;
    }
```